src/lib/authserver/dir/ds389.rb

```diff
@@ -37,29 +37,42 @@ def self.get_instance_names
   end
 
   # gen_setup_ini generates INI file content with parameters for setting up directory server.
-  def self.gen_setup_ini(fqdn, instance_name, suffix, dm_dn, dm_pass)
-    return "[General]
-FullMachineName=#{fqdn}
-SuiteSpotUserID=dirsrv
-SuiteSpotGroup=dirsrv
+  def self.gen_setup_ini(fqdn, instance_name, suffix, dm_pass)
+    return "# Generated by yast-auth-server
+[general]
+config_version = 2
+full_machine_name = #{fqdn}
+# This may be need to be tweaked, it could break setups ...
+# strict_host_checking = true/false
 
 [slapd]
-ServerPort=389
-ServerIdentifier=#{instance_name}
-Suffix=#{suffix}
-RootDN=#{dm_dn}
-RootDNPwd=#{dm_pass}
-AddSampleEntries=No
+root_password = #{dm_pass}
+instance_name = #{instance_name}
+
+[backend-userroot]
+sample_entries = yes
+suffix = #{suffix}
```

Comment:
just hint, you can use e.g. ERB files https://www.stuartellis.name/articles/erb/ to have it as separate file.

Reply:
This is too complex for me at the moment, as I don't know that much. I'd rather do a change like that in a follow up.